SConstruct

```diff
@@ -639,42 +641,40 @@ if selected_platform in platform_list:
 
     # Enforce our minimal compiler version requirements
     cc_version = methods.get_compiler_version(env) or {
-        "major": None,
-        "minor": None,
-        "patch": None,
-        "metadata1": None,
-        "metadata2": None,
-        "date": None,
+        "major": -1,
+        "minor": 0,
+        "patch": 0,
+        "metadata1": "",
+        "metadata2": "",
+        "date": "",
     }
-    cc_version_major = int(cc_version["major"] or -1)
-    cc_version_minor = int(cc_version["minor"] or -1)
-    cc_version_metadata1 = cc_version["metadata1"] or ""
+    env.compiler_version = (int(cc_version["major"]), int(cc_version["minor"]), int(cc_version["patch"]))
```

Comment:
Shouldn't this refactor be in a separate PR from the astcenc library update? Perhaps as a prerequisite?

Reply:
It is :D  #80719 
But I'm not finding the time to polish it properly.

I might update this PR and just drop the compiler-specific tweaks, to add later.